Source/WebKit/WebProcess/GPU/graphics/RemoteGraphicsContextGLProxy.cpp

```diff
@@ -289,6 +289,63 @@ void RemoteGraphicsContextGLProxy::simulateEventForTesting(SimulatedEventForTest
     }
 }
 
+void RemoteGraphicsContextGLProxy::getBufferSubData(GCGLenum target, GCGLintptr offset, std::span<uint8_t> data)
+{
+    if (isContextLost())
+        return;
+
+    if (data.empty())
+        return;
+
+    static constexpr size_t getBufferSubDataInlineSizeLimit = 64 * KB; // NOTE: when changing, change the value in RemoteGraphicsContextGL too.
+    static constexpr size_t getBufferSubDataSharedMemorySizeLimit = 100 * MB;
+
+    auto copyToData = [&](ptrdiff_t offset, std::span<const uint8_t> replyData) {
+        memcpy(data.data() + offset, replyData.data(), replyData.size_bytes());
```

Comment:
idea of std::span is that you don't anymore carry offsets and lenghts around. instead you have your span, and you cut out the unneeded parts with subspan.
Also std::copy might already be able to do the copy